CMakeLists.txt

```diff
@@ -37,6 +37,10 @@ option(OPUS_INSTALL_PKG_CONFIG_MODULE "Install PkgConfig module" ON)
 option(OPUS_INSTALL_CMAKE_CONFIG_MODULE "Install CMake package config module"
        ON)
 
+if(APPLE)
+  option(BUILD_FRAMEWORK "Build Framework bundle for Apple systems" OFF)
```

Comment:
let's use OPUS_BUILD_FRAMEWORK to keep it OPUS specific

Reply:
`BUILD_FRAMEWORK` is a CMake variable, like `BUILD_SHARED_LIBS`, it's pretty normal to add them as options like this. I thought it a bit odd that `BUILD_SHARED_LIBS` is prefixed with `OPUS_` to be honest. The Ogg CMake build doesn't prefix them.

I can change it for consistency, but it does seem odd to me to duplicate them in this way, IMO it would be better to remove the `OPUS_` prefix where providing an option to set a CMake variable.

Reply:
I see, then please duplicate similar to BUILD_SHARED_LIBS. I want people to be able to turn on and off features without affecting other projects if they are build through fetchcontent etc.

Reply:
I haven't used `FetchContent`, but with `ExternalProject_Add` you still need to explicitly pass the settings of those variables, so adding them as options directly without the prefix wouldn't interfere with other projects. The CMake documentation even states they often added directly as options: https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Reply:
> `BUILD_FRAMEWORK` is a CMake variable,

Sorry, I was wrong here, I was getting mixed up with a different variable! `BUILD_FRAMEWORK` is not a CMake variable like `BUILD_SHARED_LIBS`.